File: reports/20260826-143732/results/storage/management-plane/python/provisioning/storage-mp-python-account-mgmt/python-azure-tools/baseline/generated-code/storage_account_manager.py
"""Create, inspect, update, and delete an Azure Storage account."""

from __future__ import annotations

import os
import re
import sys
from typing import NoReturn

from azure.core.exceptions import AzureError, ClientAuthenticationError, HttpResponseError
from azure.identity import DefaultAzureCredential
from azure.mgmt.storage import StorageManagementClient
from azure.mgmt.storage.models import (
    BlobServiceProperties,
    Kind,
    Sku,
    SkuName,
    StorageAccountCreateParameters,
)


def required_environment_variable(name: str) -> str:
    value = os.getenv(name)
    if not value:
        raise ValueError(f"Required environment variable {name} is not set.")
    return value


def validate_storage_account_name(name: str) -> None:
    if not re.fullmatch(r"[a-z0-9]{3,24}", name):
        raise ValueError(
            "STORAGE_ACCOUNT_NAME must contain 3-24 lowercase letters or digits."
        )


def print_http_error(error: HttpResponseError) -> None:
    request_id = (
        error.response.headers.get("x-ms-request-id") if error.response else None
    )
    details = f" Request ID: {request_id}." if request_id else ""
    print(f"Azure request failed: {error.message}.{details}", file=sys.stderr)


def fail(message: str, exit_code: int = 1) -> NoReturn:
    print(message, file=sys.stderr)
    raise SystemExit(exit_code)


def main() -> None:
    subscription_id = required_environment_variable("AZURE_SUBSCRIPTION_ID")
    resource_group_name = required_environment_variable("RESOURCE_GROUP_NAME")
    storage_account_name = required_environment_variable("STORAGE_ACCOUNT_NAME")
    validate_storage_account_name(storage_account_name)

    credential = DefaultAzureCredential()
    client = StorageManagementClient(credential, subscription_id)

    print(f"Creating storage account {storage_account_name!r}...")
    create_poller = client.storage_accounts.begin_create(
        resource_group_name,
        storage_account_name,
        StorageAccountCreateParameters(
            sku=Sku(name=SkuName.STANDARD_LRS),
            kind=Kind.STORAGE_V2,
            location="eastus",
        ),
    )
    created_account = create_poller.result()
    print(f"Created: {created_account.id}")

    print(f"\nStorage accounts in resource group {resource_group_name!r}:")
    for account in client.storage_accounts.list_by_resource_group(
        resource_group_name
    ):
        print(f"- {account.name} ({account.location}, {account.sku.name})")

    account = client.storage_accounts.get_properties(
        resource_group_name, storage_account_name
    )
    print("\nCreated account properties:")
    print(f"  ID: {account.id}")
    print(f"  Location: {account.location}")
    print(f"  SKU: {account.sku.name}")
    print(f"  Kind: {account.kind}")
    print(f"  Provisioning state: {account.provisioning_state}")

    # Blob versioning is configured through the storage management plane's
    # blob service resource, rather than StorageAccountsOperations.update.
    print("\nEnabling blob versioning...")
    client.blob_services.set_service_properties(
        resource_group_name,
        storage_account_name,
        "default",
        BlobServiceProperties(is_versioning_enabled=True),
    )
    blob_properties = client.blob_services.get_service_properties(
        resource_group_name, storage_account_name, "default"
    )
    print(f"Blob versioning enabled: {blob_properties.is_versioning_enabled}")

    print(f"\nDeleting storage account {storage_account_name!r}...")
    client.storage_accounts.delete(resource_group_name, storage_account_name)
    print("Storage account deleted.")


if __name__ == "__main__":
    try:
        main()
    except ValueError as error:
        fail(f"Configuration error: {error}", 2)
    except ClientAuthenticationError as error:
        fail(f"Azure authentication failed: {error}", 3)
    except HttpResponseError as error:
        print_http_error(error)
        raise SystemExit(4) from error
    except AzureError as error:
        fail(f"Azure SDK error: {error}", 5)
    except KeyboardInterrupt:
        fail("Operation cancelled.", 130)

File: rerun-reports/lane-a/043/20260828-052218/results/storage/management-plane/java/provisioning/storage-mp-java-account-mgmt/java-azure-skills/azure-skill-mcp/generated-code/src/main/java/example/StorageAccountManager.java
package example;

import com.azure.core.credential.TokenCredential;
import com.azure.core.exception.ClientAuthenticationException;
import com.azure.core.exception.HttpResponseException;
import com.azure.core.management.Region;
import com.azure.core.management.profile.AzureProfile;
import com.azure.core.models.AzureCloud;
import com.azure.identity.DefaultAzureCredentialBuilder;
import com.azure.resourcemanager.storage.StorageManager;
import com.azure.resourcemanager.storage.models.BlobServiceProperties;
import com.azure.resourcemanager.storage.models.StorageAccount;
import com.azure.resourcemanager.storage.models.StorageAccountSkuType;

public final class StorageAccountManager {
    private static final Region REGION = Region.US_EAST;

    private StorageAccountManager() {
    }

    public static void main(String[] args) {
        if (args.length != 2) {
            System.err.println(
                "Usage: mvn exec:java -Dexec.args=\"<resource-group> <globally-unique-storage-account-name>\"");
            System.exit(2);
        }

        String resourceGroupName = args[0];
        String storageAccountName = args[1];
        String subscriptionId;

        try {
            subscriptionId = requireEnvironmentVariable("AZURE_SUBSCRIPTION_ID");
            validateStorageAccountName(storageAccountName);
        } catch (IllegalArgumentException exception) {
            System.err.println("Invalid configuration: " + exception.getMessage());
            System.exit(2);
            return;
        }

        StorageManager storageManager = null;
        boolean accountCreated = false;
        int exitCode = 0;

        try {
            storageManager = createStorageManager(subscriptionId);

            StorageAccount createdAccount = storageManager.storageAccounts()
                .define(storageAccountName)
                .withRegion(REGION)
                .withExistingResourceGroup(resourceGroupName)
                .withSku(StorageAccountSkuType.STANDARD_LRS)
                .withGeneralPurposeAccountKindV2()
                .withOnlyHttpsTraffic()
                .create();
            accountCreated = true;
            System.out.printf("Created storage account: %s%n", createdAccount.id());

            System.out.printf("Storage accounts in resource group '%s':%n", resourceGroupName);
            for (StorageAccount account
                : storageManager.storageAccounts().listByResourceGroup(resourceGroupName)) {
                System.out.printf("  %s (%s)%n", account.name(), account.regionName());
            }

            StorageAccount account = storageManager.storageAccounts()
                .getByResourceGroup(resourceGroupName, storageAccountName);
            if (account == null) {
                throw new IllegalStateException(
                    "The created storage account could not be retrieved.");
            }

            System.out.println("Created storage account properties:");
            System.out.printf("  ID: %s%n", account.id());
            System.out.printf("  Name: %s%n", account.name());
            System.out.printf("  Region: %s%n", account.regionName());
            System.out.printf("  SKU: %s%n", account.skuType());
            System.out.printf("  Kind: %s%n", account.kind());

            BlobServiceProperties blobProperties = storageManager.blobServices()
                .getServicePropertiesAsync(resourceGroupName, storageAccountName)
                .block();
            if (blobProperties == null) {
                throw new IllegalStateException(
                    "The storage account's Blob service properties could not be retrieved.");
            }

            BlobServiceProperties updatedBlobProperties = blobProperties.update()
                .withBlobVersioningEnabled()
                .apply();
            if (!Boolean.TRUE.equals(updatedBlobProperties.isBlobVersioningEnabled())) {
                throw new IllegalStateException("Blob versioning was not enabled.");
            }
            System.out.println("Blob versioning enabled.");
        } catch (ClientAuthenticationException exception) {
            System.err.println("Azure authentication failed: " + exception.getMessage());
            exitCode = 1;
        } catch (HttpResponseException exception) {
            printHttpError("Azure Storage management operation failed", exception);
            exitCode = 1;
        } catch (IllegalArgumentException | IllegalStateException exception) {
            System.err.println("Storage account operation failed: " + exception.getMessage());
            exitCode = 1;
        } finally {
            if (accountCreated && storageManager != null) {
                try {
                    storageManager.storageAccounts()
                        .deleteByResourceGroup(resourceGroupName, storageAccountName);
                    System.out.printf("Deleted storage account: %s%n", storageAccountName);
                } catch (ClientAuthenticationException exception) {
                    System.err.println(
                        "Authentication failed while deleting the storage account: "
                            + exception.getMessage());
                    exitCode = 1;
                } catch (HttpResponseException exception) {
                    printHttpError("Failed to delete the storage account", exception);
                    exitCode = 1;
                }
            }
        }

        if (exitCode != 0) {
            System.exit(exitCode);
        }
    }

    private static StorageManager createStorageManager(String subscriptionId) {
        AzureProfile profile = new AzureProfile(
            null,
            subscriptionId,
            AzureCloud.AZURE_PUBLIC_CLOUD);

        TokenCredential credential = new DefaultAzureCredentialBuilder()
            .authorityHost(profile.getEnvironment().getActiveDirectoryEndpoint())
            .build();

        return StorageManager.authenticate(credential, profile);
    }

    private static String requireEnvironmentVariable(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException(
                "Environment variable " + name + " must be set.");
        }
        return value;
    }

    private static void validateStorageAccountName(String name) {
        if (!name.matches("[a-z0-9]{3,24}")) {
            throw new IllegalArgumentException(
                "The storage account name must contain 3-24 lowercase letters or digits.");
        }
    }

    private static void printHttpError(String context, HttpResponseException exception) {
        String status = exception.getResponse() == null
            ? "unknown"
            : Integer.toString(exception.getResponse().getStatusCode());
        System.err.printf("%s (HTTP %s): %s%n", context, status, exception.getMessage());
    }
}
